### PR TITLE
ci: automated artefact publish to maven central

### PR DESCRIPTION
**Pull request by SrinivasanTarget**
## Change list

Publish artifacts to maven central on release
 
## Types of changes

What types of changes are you proposing/introducing to Java client?
_Put an `x` in the boxes that apply_

- [ ] No changes in production code.
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up Java
+        uses: actions/setup-java@v3
+        with:
+          java-version: '11'
+          distribution: 'adopt'
+      - name: Validate Gradle wrapper
+        uses: gradle/wrapper-validation-action@e6e38bacfdf1a337459f332974bb2327a31aaf4b
```
Comment (valfirst) [inline .github/workflows/publish.yml:16]: is it required for release workflow?

File: .github/workflows/publish.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+      - name: Publish package
+        uses: gradle/gradle-build-action@67421db6bd0bf253fb4bd25b31ebb98943c375e1
+        with:
+          arguments: publish
+        env:
+          MAVEN_USERNAME: ${{ secrets.OSSRH_USERNAME }}
+          MAVEN_PASSWORD: ${{ secrets.OSSRH_TOKEN }}
```
Comment (valfirst) [inline .github/workflows/publish.yml:23]: maybe just:
```suggestion
      - name: Publish package
        env:
          MAVEN_USERNAME: ${{ secrets.OSSRH_USERNAME }}
          MAVEN_PASSWORD: ${{ secrets.OSSRH_TOKEN }}
        run: ./gradlew publish

```
?

File: .github/workflows/publish.yml [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    types: [created]
+jobs:
+  publish:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up Java
+        uses: actions/setup-java@v3
+        with:
+          java-version: '11'
```
Comment (valfirst) [inline .github/workflows/publish.yml:13]: I would use `8` to be 100% sure the release will be JDK 8 compatible

**Review by valfirst — CHANGES_REQUESTED** [review]

File: .github/workflows/publish.yml (hunk 1 above)
Comment (SrinivasanTarget) [reply .github/workflows/publish.yml:16]: Good catch. It's not required here as we already have a separate workflow that only validates the wrapper.

File: .github/workflows/publish.yml [hunk 4]
```diff
@@ -0,0 +1,23 @@
+name: Publish package to the Maven Central Repository
+on:
+  release:
+    types: [created]
```
[the remark this answers is not in the document]
Comment (SrinivasanTarget) [reply .github/workflows/publish.yml:4]: As soon as the release is published in GitHub this workflow will get triggered automatically.

**Review by valfirst — APPROVED** [review]

**Comment by SrinivasanTarget** [comment]
@Higgens75 i don't see any problem with publishing as I see it is available [here](https://central.sonatype.dev/artifact/io.appium/java-client/8.3.0)

**Pull request merged**